pennylane/drawer/tape_mpl.py

```diff
@@ -271,19 +256,74 @@ def _tape_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, *, fig
         for op in layer_ops:
             _add_operation_to_drawer(op, drawer, layer, config)
 
-    for wire in _get_measured_wires(tape.measurements, list(range(n_wires))):
-        drawer.measure(n_layers, wire)
+    if measurements == "dots":
+        for wire in wire_map.values():
+            drawer.ax.text(
+                n_layers + 0.25,
+                wire,
+                s="...",
+                ha="center",
+                va="center_baseline",
+                fontsize=21,
+            )
+    else:
+        for wire in _get_measured_wires(measurements, list(range(n_wires))):
+            drawer.measure(n_layers, wire)
 
-    measured_bits = _get_measured_bits(tape.measurements, bit_map, drawer.n_wires)
-    if measured_bits:
-        drawer.measure(n_layers, measured_bits)
+        measured_bits = _get_measured_bits(measurements, bit_map, drawer.n_wires)
+        if measured_bits:
+            drawer.measure(n_layers, measured_bits)
 
     return drawer.fig, drawer.ax
 
 
+def _tape_mpl(tape, wire_order=None, show_all_wires=False, max_length=None, **kwargs):
+    """Internal implementation for matplotlib circuit visualization.
+
+    Handles the core drawing logic while the public tape_mpl wrapper manages styling.
+
+    Returns:
+        When max_length is None: tuple(Figure, Axes) for the complete circuit
+        Otherwise: list[tuple(Figure, Axes)] with one pair per circuit segment
+    """
+    wire_map = convert_wire_order(tape, wire_order=wire_order, show_all_wires=show_all_wires)
+    tape = transform_deferred_measurements_tape(tape)
+    tape = qml.map_wires(tape, wire_map=wire_map)[0][0]
+    bit_map = default_bit_map(tape)
+
+    layers = drawable_layers(tape.operations, wire_map={i: i for i in tape.wires}, bit_map=bit_map)
+
+    for i, layer in enumerate(layers):
+        if any(isinstance(o, qml.measurements.MidMeasureMP) and o.reset for o in layer):
+            layers.insert(i + 1, [])
+
+    if max_length is None:
+        return _draw_layers(layers, tape.measurements, bit_map=bit_map, wire_map=wire_map, **kwargs)
+
+    layer_count = len(layers)
+    return [
+        _draw_layers(
+            layers=layers[i : i + max_length],
+            measurements=tape.measurements if i + max_length >= layer_count else "dots",
+            bit_map=bit_map,
+            wire_map=wire_map,
+            **kwargs,
+        )
+        for i in range(0, layer_count, max_length)
+    ]
+
+
 # pylint: disable=too-many-arguments
 def tape_mpl(
-    tape, wire_order=None, show_all_wires=False, decimals=None, style=None, *, fig=None, **kwargs
+    tape,
+    wire_order=None,
+    show_all_wires=False,
+    decimals=None,
+    style=None,
+    *,
+    fig=None,
+    max_length: Optional[int] = None,
+    **kwargs,
```

Comment:
```suggestion
    tape: QuantumScript,
    wire_order: Optional[Sequence] = None,
    show_all_wires: bool = False,
    decimals: Optional[int] = None,
    style: Optional[str] = None,
    *,
    fig: Optional["mpl.figure.Figure"] = None,
    max_length: Optional[int] = None,
    **kwargs: Any,
```

Or something similar. We would also need to add a statement in the docstring of the function to reflect the new kwarg, something like:

```
        max_length (int): The maximum number of layers to draw in a single figure. If the circuit
            exceeds this number, it will be split into multiple figures.
```

or whatever you prefer : )

Reply:
type hints added.

Reply:
LGTM. Happy to approve once tests pass 🚀